index.js

```diff
@@ -404,6 +404,64 @@ class Encore {
         return this;
     }
 
+    /**
+     * Copy files or folders to the build directory.
+     *
+     * For example:
+     *
+     *      // Copy the content of a whole directory and its subdirectories
+     *      Encore.copyFiles({ from: './images' });
+     *
+     *      // Only copy files matching a given pattern
+     *      Encore.copyFiles({ from: './images', pattern: /\.(png|jpg|jpeg)$/ })
+     *
+     *      // Set the path the files are copied to
+     *      Encore.copyFiles({
+     *          from: './images',
+     *          pattern: /\.(png|jpg|jpeg)$/,
+     *          to: 'assets/images/[path][name].[ext]'
+     *      })
+     *
+     *      // Version files
+     *      Encore.copyFiles(
+     *          from: './images',
+     *          to: 'assets/images/[path][name].[hash:8].[ext]'
+     *      })
+     *
+     *      // Add multiple configs in a single call
+     *      Encore.copyFiles([
+     *          { from: './images' },
+     *          { from: './txt', pattern: /\.txt$/ },
+     *      ]);
+     *
+     * Notes:
+     *      * No transformation is applied to the copied files (for instance
+     *        copying a CSS file won't minify it)
+     *      * By default files won't be versioned even if enableVersioning()
+     *        has been called. In order to add a hash to your filenames you
+     *        must use the "[hash]" placeholder in the "to" option (see below)
```

Comment:
This part surprised me. Though, I can see now (I think) why this is the case: we're parsing everything through file-loader, and the user is in control of their filename. 2 questions:

1) Without ever writing any code, I was expecting the implementation of this would be basically to loop over and `require` each file that you want copied. In that case, they *would* be versioned automatically. But I guess, the problem is that you don't have control over the final filename / path?

2) Could we (or is it a bad idea) hack the `[hash]` in when it's versioned? Or, do this if `to` is not specified, but make it the responsibility of the user if it *is* specified?

Reply:
Actually the issue comes from the way the different calls to `Encore.copyFiles` are currently saved.

If you look at the `copyFiles` function in `WebpackConfig.js` you'll see that the `to` property of the `defaultConfig` object contains the default pattern (`[path][name].[ext]`). Unless the user explicitly set `to` when adding a config it'll be this pattern that will be used when generating the related `file-loader` rule.

Since we don't know at that point if the versioning is enabled  it's not possible to replace it by a versioned pattern.

That being said, if you want it to works that way I can probably do it quite easily by setting the `to` option to `null` in the default config. And then choose which pattern should be used only when generating the `require` calls.